spec/ruby-push-notifications/apns/apns_notification_spec.rb

```diff
@@ -16,6 +16,7 @@ module APNS
       end
 
       it 'caches the payload' do
+        allow(data).to receive(:respond_to?).and_return(false)
```

Comment:
Allowing to receive `respond_to` is probably too generic. Could you please add `with(:to_json)` to limit the stubbing?